src/components/NewDatePicker/index.js

```diff
@@ -65,9 +67,14 @@ class NewDatePicker extends React.Component {
     }
 
     /**
-     * Function to animate and hide the picker.
+     * Function to animate and hide the picker when the user clicks outside of it.
+     * @param {Event} event - The event that triggered the hide action.
      */
-    hidePicker() {
```

Comment:
`hidePicker` should take no parameter. Handle the `event` on the `onBlur` callback i.e.:
```js
onBlur={(event) => {
    if (return early condition is met) {
        return;
    }
    hidePicker();
}}
```